src/Form/Control/Password.php

```diff
@@ -4,7 +4,88 @@
 
 namespace Atk4\Ui\Form\Control;
 
+use Atk4\Ui\Icon;
+use Atk4\Ui\Js\JsReload;
+
 class Password extends Line
 {
     public string $inputType = 'password';
+
+    /** Enable password reveal */
+    public bool $revealEye = false;
+
+    /** @var Icon|array|null */
+    private $revealEyeIcon;
+
+    #[\Override]
+    protected function init(): void
+    {
+        parent::init();
+
+        if ($this->revealEye) {
+            if (!$this->icon) {
+                $this->icon = 'true';
+            }
+        }
+    }
+
+    /**
+     * Remove selected value/values from array.
+     *
+     * @return array
+     */
+    protected function delArrValues(array $arr, array $remove)
+    {
+        return array_filter($arr, static fn ($e) => !in_array($e, $remove, true));
+    }
+
+    #[\Override]
+    protected function recursiveRender(): void
+    {
+        if ($this->revealEye) {
+            if ($this->stickyGet('iconInit')) {
+                $iconInit = $this->stickyGet('iconInit');
```

Comment:
What is the motivation to persist reveal across loads?

Reply:
When the click event occurs 2 Jqueries are sent, one POST, which is ok, the second is a GET, this resets the class of the field to default.
To fix this I send the class field and the inputType so it is persistant after the GET.

Reply:
reverted in https://github.com/atk4/ui/pull/2219/commits/d7cf0494bffbd06ad7ff8146604bed100bba9e74